src/main/java/Entities/Playlist.java

```diff
@@ -17,13 +17,12 @@ public Playlist(String name, User owner, boolean isPublic){
         this.name = name;
         this.owner = owner;
         this.isPublic = isPublic;
-        this.songList = new LinkedList<>();
+        this.songList = new LinkedList<Song>();
     }
     public Playlist(String name, User owner, boolean isPublic, Song firstSong){
         this.name = name;
         this.owner = owner;
         this.isPublic = isPublic;
-        this.songList = new LinkedList<>();
```

Comment:
Autograder fails because you deleted this line - you have to instantiate the list before calling addFirst().